kernel/kernel-api/src/main/java/io/delta/kernel/internal/deletionvectors/RoaringBitmapArray.java

```diff
@@ -258,17 +259,39 @@ private void extendBitmaps(int newLength) {
         newBitmaps, // dest
         0, // dest start pos
         bitmaps.length); // number of entries to copy
-    for (int i = 0; i < bitmaps.length; i++) {
+    for (int i = bitmaps.length; i < newBitmaps.length; i++) {
       newBitmaps[i] = new RoaringBitmap();
     }
     bitmaps = newBitmaps;
   }
 
   public static RoaringBitmapArray create(long... values) {
     RoaringBitmapArray bitmap = new RoaringBitmapArray();
+    bitmap.bitmaps = new RoaringBitmap[0];
```

Comment:
Instead of doing this here we should just initialize `bitmaps` to an empty array in the class. Since we don't have a constructor you can just do this on line 103

Reply:
👍  Good point